src/meta/typesmap.go

```diff
@@ -376,3 +376,31 @@ func (m TypesMap) Iterate(cb func(typ string)) {
 		cb(k)
 	}
 }
+
+// Checks that the map contains only arrays.
+// Not for *Lazy* type.
+func (m TypesMap) ContainsOnlyArrays() bool {
```

Comment:
You can probably just ignore the types that are not arrays. In PHP it can be really hard to know the types for sure anyway so it's also hard to guarantee that the type will only be an array and nothing else.